hw3 pull request

IhorStetskyi opened the pull request.

IhorStetskyi committed:
changed

Hekitkoo commented on Exception Handling/Task1/bin/Debug/netcoreapp3.1/Task1.deps.json:
```diff
@@ -0,0 +1,23 @@
+{
```
@IhorStetskyi, for the following tasks, please add default .gitignore for the .net project. That help clearly reviews your code.

IhorStetskyi replied:
gitignore added

Hekitkoo commented on Exception Handling/Task1/Program.cs:
```diff
@@ -1,10 +1,34 @@
     {
         private static void Main(string[] args)
         {
             // TODO: Implement the task here.
```
@IhorStetskyi, please remove comments that have already been fixed.

IhorStetskyi replied:
removed old comments

Hekitkoo commented on Exception Handling/Task2/NumberParser.cs:
```diff
@@ -6,7 +6,100 @@ public class NumberParser : INumberParser
+                for (int i = startChar; i < n; i++)
+                {
+                    // Subtract 48 from the current digit 
+                    int tempnum = num * 10 + ((int)stringValue[i] - 48);
```
@IhorStetskyi, extract magic number (48 - it's '0' so you could extract it as ZeroCode variable)

IhorStetskyi replied:
number 48 is extracted into ZeroCode variable

Hekitkoo commented on Exception Handling/Task2/NumberParser.cs:
```diff
@@ -6,7 +6,100 @@ public class NumberParser : INumberParser
+            }
+        }
+
+        int SelectStartChar(string s)
```
@IhorStetskyi, you already have sign char, we could pass it here.

Hekitkoo commented on Exception Handling/Task2/NumberParser.cs:
```diff
@@ -6,7 +6,100 @@ public class NumberParser : INumberParser
+                {
+                    throw new FormatException("Wrong format");
+                }
+                if (i > 0 && !"1234567890".Contains(someString[i]))
```
@IhorStetskyi magic string, please extract to a variable.

IhorStetskyi replied:
magic someString changed to InputDataString

Hekitkoo commented on Exception Handling/Task3/UserTaskService.cs:
```diff
@@ -12,25 +13,49 @@ public UserTaskService(IUserDao userDao)
-                if (string.Equals(task.Description, t.Description, StringComparison.OrdinalIgnoreCase))
-                    return -3;
-            }
+                if (userId < 0)
```
@IhorStetskyi extract validation to the method, please. [Single responsibility](https://en.wikipedia.org/wiki/Single-responsibility_principle) good explains why should do it.

IhorStetskyi replied:
Additional CheckForStringEqualityIgnoringCase, CheckForException, TryToCreateTask methods created to meet Single responsibility principle

Hekitkoo reviewed (changes requested):
Good work, but need some changes.

IhorStetskyi committed:
.added gitignore to Exception handling

IhorStetskyi committed:
Class:Exception Handling/Task2/NumberParser.cs:
-removed old comments
-number 48 is extracted into ZeroCode variable
-magic someString changed to InputDataString

Class:Handling/Task3/UserTaskService.cs:
-Additional CheckForStringEqualityIgnoringCase, CheckForException, TryToCreateTask methods created to meet Single responsibility principle
